src/pages/RFOX/components/Claim/Claim.tsx

```diff
@@ -44,10 +49,28 @@ export const Claim: React.FC<ClaimRouteProps> = ({ headerComponent, setStepIndex
 
 export const ClaimRoutes: React.FC<ClaimRouteProps> = ({ headerComponent, setStepIndex }) => {
   const location = useLocation()
+  const queryClient = useQueryClient()
 
   const [confirmedQuote, setConfirmedQuote] = useState<RfoxClaimQuote | undefined>()
   const [claimTxid, setClaimTxid] = useState<string | undefined>()
 
+  const { queryKey: unstakingRequestCountQueryKey } = useGetUnstakingRequestCountQuery({
+    stakingAssetAccountAddress: confirmedQuote
+      ? getAddress(fromAccountId(confirmedQuote.stakingAssetAccountId).account)
+      : undefined,
+  })
+
+  const { queryKey: unstakingRequestQueryKey } = useGetUnstakingRequestsQuery({
+    stakingAssetAccountAddress: confirmedQuote
+      ? getAddress(fromAccountId(confirmedQuote.stakingAssetAccountId).account)
+      : undefined,
+  })
+
+  const handleTxConfirmed = useCallback(async () => {
+    await queryClient.invalidateQueries({ queryKey: unstakingRequestCountQueryKey })
+    await queryClient.invalidateQueries({ queryKey: unstakingRequestQueryKey })
```

Comment:
preferably blocking: If the intent here is only to invalidate the corresponding queries matching the query key, it could be lighter on the app to grab the query key only above without invoking the query via `useGetUnstakingRequestCountQuery` and `useGetUnstakingRequestsQuery`, otherwise we can have a case where we invoke the query, and then invalidate it immediately

An example of the pattern to expose query key getters is in `src/pages/RFOX/hooks/useEpochHistoryQuery.ts` where a `getEpochHistoryQueryKey` is a separate function to the hook itself, allowing you to invalidate a query without invoking it in the case none matches the key you generated, reducing overhead in that case.

Reply:
Generally would agree with that, though those queries are a bit particular ones:

What we do with those wagmi wrappers is effectively reconstructing wagmi hooks (e.g `useReadContract`) in a type-safe way (i.e supporting `skipToken`), using the same core utils as them and having matching `queryKey`s to be aligned with their implementation, meaning when wagmi bumps their react-query version to v5, we should be able to ditch our hand-rolled hooks, and simply consume the wagmi variant.

<img width="573" alt="image" src="https://github.com/shapeshift/web/assets/17035424/457b559f-0a26-42de-8a6a-61fa374d1686">

we could make the queryKey-building a function instead of an expression, however there is one main issue with that:

If (hopefully wen, doubt wagmi is going to be stuck on v4 forever) wagmi bumps to v5 and we ditch our hand-rolled hooks, then we're good-to-go with the current approach as [this is also how wagmi does it](https://wagmi.sh/react/guides/migrate-from-v1-to-v2#removed-watch-property) - however, if we were to implement some query-buliding function instead, then we wouldn't be able to remove our hand-rolled hooks.

tl;dr everytime we see some hook that's effectively our hand-rolled wagmi hook remade from source, just assume that the API is similar to wagmi's and that we don't have access to internals in a way that would let us e.g build the queryKey in isolation

Reply:
Ah sure that makes sense in terms of formulating a drop-in replacement so we can upgrade to wagmi v5 semlessly, though the question remains:
If the query for the given key has not fired, does us invalidating it cause it to fetch then invalidate then fetch again, effectively 2 fetches?
I'm going to stamp so we can get this moving as it seems more of a theoretical concern